Prepare X-Net

liuchengxu opened the pull request:
The only notable difference between Gemini and X-Net is that X-Net enables the executor from the genesis, the others are mostly about the names.

liuchengxu committed:
Prepare testnet for executor

nazar-pc reviewed (dismissed):
Maybe runtime version bump here?

liuchengxu committed:
Bump runtime version to 3

liuchengxu commented on crates/subspace-node/src/chain_spec.rs:
```diff
@@ -167,6 +167,101 @@ pub fn gemini_config_compiled(
 }
 
+pub fn x_net_config() -> Result<ConsensusChainSpec<GenesisConfig, ExecutionGenesisConfig>, String> {
+    todo!("Finalize the X-net ChainSpec")
```
Will merge after this is resolved at least.

liuchengxu committed:
Build raw chain spec for X-Net

It's built locally for simplicity.

```bash
./target/release/subspace-node build-spec --disable-default-bootnode --chain x-net-1-compiled --raw > chain-spec-raw-x-net-1.json
```

liuchengxu committed:
Merge branch 'main' into x-net-config
